app/server/server.go

```diff
@@ -41,12 +48,26 @@ func CreateClientOptionsFromEnv() (client.Options, error) {
 
 	if namespaceName == "" {
 		namespaceName = "default"
-		fmt.Printf("Namespace name unspecified; using value '%s'\n", namespaceName)
+		slog.Warn("Namespace name unspecified; using default", "namespace", namespaceName)
 	}
 
 	clientOpts := client.Options{
 		HostPort:  hostPort,
 		Namespace: namespaceName,
+		Logger:    tlog.NewStructuredLogger(slog.Default()),
+	}
+
+	if os.Getenv("PROMETHEUS_ENDPOINT") != "" {
+		exporter, err := prometheus.New()
+		if err != nil {
+			return client.Options{}, err
+		}
+		provider := metric.NewMeterProvider(metric.WithReader(exporter))
+		meter := provider.Meter(meterName)
+
+		clientOpts.MetricsHandler = opentelemetry.NewMetricsHandler(
```

Comment:
Interesting choice to use OTel instead of Tally. Not that I disagree, but can we confirm via end-to-end tests that it is working properly? Should be easy to just make an http call to the metrics endpoint and check it.